### ISSUE TITLE
Rails5: Boolean column gets returned as true when the column value is 'N' when using emulate_booleans_from_strings

### ISSUE THREAD

**Issue by awaltman** [rsim/oracle-enhanced#979]
Hi Yasuo,

I think I discovered an issue when using Rails 5 and the Oracle Enhanced adapter with emulate_booleans_from_strings = true. Rows appear to be entered into the table correctly, but when they are fetched, rows that have the value 'N' for the boolean column have the value true in the model instance. Could you please take a look at the example and let me know? I also wondered if it is related to the spec failures issue (https://github.com/rsim/oracle-enhanced/issues/942).

Thanks!
Andrew

Environment:
CRuby 2.3.1
Rails: 5.0.0.1
Oracle Enhanced Adapter: latest from github (at master@e1ff4f5)

Migration:
```ruby
  def self.up
    create_table "test_bools", force: true, primary_key_trigger: true, :sequence_start_value =>1000001 do |t|
        t.string  "name",                       null: false
        t.boolean "is_default", default: "N"
    end

    add_index "test_bools", ["name"], name: "test_bs_uniq_name_key_index", unique: true
  end
```
Example using Rails console:
```ruby
$ RAILS_ENV=development bundle exec rails c
Loading development environment (Rails 5.0.0.1)
2.3.1 :001 > t = TestBool.create(name: 'Test 1', is_default: false)
   (5.0ms)  SELECT cols.column_name AS name, cols.data_type AS sql_type, cols.data_default, cols.nullable, cols.virtual_column, cols.hidden_column, cols.data_type_owner AS sql_type_owner, DECODE(cols.data_type, 'NUMBER', data_precision, 'FLOAT', data_precision, 'VARCHAR2', DECODE(char_used, 'C', char_length, data_length), 'RAW', DECODE(char_used, 'C', char_length, data_length), 'CHAR', DECODE(char_used, 'C', char_length, data_length), NULL) AS limit, DECODE(data_type, 'NUMBER', data_scale, NULL) AS scale, comments.comments as column_comment FROM all_tab_cols cols, all_col_comments comments WHERE cols.owner = 'RS' AND cols.table_name = 'TEST_BOOLS' AND cols.hidden_column = 'NO' AND cols.owner = comments.owner AND cols.table_name = comments.table_name AND cols.column_name = comments.column_name ORDER BY cols.column_id
  Sequence (1.0ms)  select us.sequence_name from all_sequences us where us.sequence_owner = 'RS' and us.sequence_name = 'TEST_BOOLS_SEQ'
  Primary Key (1.8ms)  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'RS' AND c.table_name = 'TEST_BOOLS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name
  Sequence (0.8ms)  select us.sequence_name from all_sequences us where us.sequence_owner = 'RS' and us.sequence_name = 'TEST_BOOLS_SEQ'
  Primary Key (1.2ms)  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'RS' AND c.table_name = 'TEST_BOOLS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name
  Primary Key Trigger (35.8ms)            SELECT trigger_name
          FROM all_triggers
          WHERE owner = 'RS'
            AND trigger_name = 'TEST_BOOLS_PKT'
            AND table_owner = 'RS'
            AND table_name = 'TEST_BOOLS'
            AND status = 'ENABLED'

  SQL (48.2ms)  INSERT INTO "TEST_BOOLS" ("NAME") VALUES (:a1) RETURNING "ID" INTO :returning_id  [["name", "Test 1"], ["returning_id", nil]]
 => #<TestBool id: 1000001, name: "Test 1", is_default: false>
2.3.1 :002 > t2 = TestBool.find(1000001)
  TestBool Load (3.6ms)  SELECT  "TEST_BOOLS".* FROM "TEST_BOOLS" WHERE "TEST_BOOLS"."ID" = :a1 FETCH FIRST :a2 ROWS ONLY  [["id", 1000001], ["LIMIT", 1]]
 => #<TestBool id: 1000001, name: "Test 1", is_default: true>
2.3.1 :003 > c = OCI8.new('rs',<password>,'xe')
 => #<OCI8:RS>
2.3.1 :004 > c.exec('select * from test_bools') do |r| puts r.join(','); end
1000001,Test 1,N
 => 1
```

**Comment by yahonda** [comment]
Thanks for using Oracle enhanced adapter 1.7 with Rail 5 and report the issue.

I think I am able to reproduce it.
```ruby

begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  gem 'rails', '= 5.0.0.1'
  gem 'activerecord', '= 5.0.0.1'
  gem 'activerecord-oracle_enhanced-adapter', '~> 1.7.0'
  gem 'ruby-oci8'
  gem 'minitest'
  gem 'byebug'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'
require 'active_record/connection_adapters/oracle_enhanced_adapter'
require 'byebug'

# Ensure backward compatibility with Minitest 4
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)

# Set Oracle enhanced adapter specific connection parameters
DATABASE_NAME = ENV['DATABASE_NAME'] || 'orcl'
DATABASE_HOST = ENV['DATABASE_HOST']
DATABASE_PORT = ENV['DATABASE_PORT']
DATABASE_USER = ENV['DATABASE_USER'] || 'oracle_enhanced'
DATABASE_PASSWORD = ENV['DATABASE_PASSWORD'] || 'oracle_enhanced'
DATABASE_SYS_PASSWORD = ENV['DATABASE_SYS_PASSWORD'] || 'admin'

CONNECTION_PARAMS = {
  :adapter => "oracle_enhanced",
  :database => DATABASE_NAME,
  :host => DATABASE_HOST,
  :port => DATABASE_PORT,
  :username => DATABASE_USER,
  :password => DATABASE_PASSWORD
}

#ActiveRecord::Base.default_timezone = :local
ActiveRecord::Base.default_timezone = :utc
ActiveRecord::Base.establish_connection(CONNECTION_PARAMS)

ActiveRecord::Base.logger = Logger.new(STDOUT)
ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter.emulate_booleans_from_strings = true

ActiveRecord::Schema.define do
  create_table :posts, :force => true do |t|
    t.string  "name",                       null: false
    t.boolean "is_default", default: "N"
  end
end

class Post < ActiveRecord::Base
end

class BugTest < Minitest::Test
  def test_is_default_after_reload
    post = Post.create(name: 'Test 1', is_default: false)
    assert_equal  false, post.is_default
    post.reload
    assert_equal  false, post.is_default
  end
end
```

* Result
```ruby


  1) Failure:
BugTest#test_is_default_after_reload [reprails50.rb:66]:
Expected: false
  Actual: true

1 runs, 2 assertions, 1 failures, 0 errors, 0 skips
```

* Environment
```
- ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-linux]
- Oracle Database 12c Enterprise Edition Release 12.1.0.2.0 - 64bit Production
```

**Comment by yahonda** [comment]
@awaltman Would you add this entry to your Gemfile and see if this issue still reproduce or not.

```ruby
gem 'activerecord-oracle_enhanced-adapter', git: 'git@github.com:yahonda/oracle-enhanced.git', branch: 'diag979_master'
```

**Comment by awaltman** [comment]
@yahonda I tried the code from that branch and it fixes the issue in my application. 

Thank you for the very quick response and diligent work on the adapter!

Andrew

**Comment by yahonda** [comment]
Thanks for the testing and glad your issue resolved. Will add some tests and release a new version of Oracle enhanced adapter within some couple of days.

**Comment by yahonda** [comment]
Closed as #985 merged to master and released v1.7.1 including this fix.
